### PR TITLE
chore(operations): Build archives, DEB, and RPM packages with glibc

### LINKED ISSUE #2337: Update our release process to favor glibc builds
Given the results in #2313, we should update our release process to produce Vector versions compiled with glibc. Here's what that looks like:

### Strategy

We should _not_ rip out our MUSL builds, instead, we should continue to offer MUSL builds as an option. While offering both MUSL and Glibc builds can be confusing to users, we can eliminate that confusion by making the MUSL builds only discoverable by browsing packages.timber.io/vector. Pages like [https://vector.dev/releases/0.8.2/download/](https://vector.dev/releases/0.8.2/download/) do not need to display MUSL based download links, and we can mention that MUSL builds are available on pages like [https://vector.dev/docs/setup/installation/manual/from-archives/](https://vector.dev/docs/setup/installation/manual/from-archives/).

I like offering both because:

1. Some users require this depending on their environment.
2. Some users might prefer this.
3. Finally, to preserve backward compatibility with existing users.

#### Preface

- [ ] Add a new builder Docker image that dynamically links to `glibc` 2.19 and statically links to `libc++` or `libstdc++`. `libc` is needed for `leveldb`. Blocked by #2336. https://github.com/timberio/vector/pull/2328 represents the initial work.

#### Archives

- [ ] Update the `vector-x86_64-unknown-linux-gnu.tar.gz` archive to use the new `glibc` based build. Currently, this is just a redirect to `vector-x86_64-unknown-linux-musl.tar.gz`.
- [ ] Produce a new `vector-aarch64-unknown-linux-gnu.tar.gz` archive. (is this necessary)
- [ ] Produce a new `vector-armv7-unknown-linux-gnueabihf.tar.gz`. (is this necessary)
- [ ] Update the `vector-x86_64-apple-darwin.tar.gz` archive to use the new `glibc` based build.

#### Packages

- [ ] Update the Vector `.deb` packages to use the new `glibc` based build.
- [ ] Update the Vector `.rpm` packages to use the new `glibc` based build.

#### Docker

- [ ] Update the Vector Docker `*-debian` variant to use the new `glibc` based build.
- [X] The Vector Docker `*-alpine` variant should continue to use MUSL since this is required for Alpine.

#### Postface

- [ ] 🎉

### PR DESCRIPTION
**Pull request by a-rodin**
Closes #2337.

### CODE REVIEW

File: scripts/ci-docker-images/builder-x86_64-unknown-linux-gnu/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# The official Rust image based on Debian 8 Jessie.
```
Comment (binarylogic) [inline scripts/ci-docker-images/builder-x86_64-unknown-linux-gnu/Dockerfile:1]: This file seems much much simpler 👍. Are we statically linking `leveldb` and `rdkafka`?

Comment (a-rodin) [reply scripts/ci-docker-images/builder-x86_64-unknown-linux-gnu/Dockerfile:1]: > Are we statically linking leveldb and rdkafka?

Yes!

File: docker-compose.yml [hunk 2]
```diff
@@ -22,14 +22,31 @@ services:
...
+      NATIVE_BUILD: "false"
+      TARGET: x86_64-unknown-linux-gnu
+      FEATURES: default
+      CARGO_TERM_COLOR: always
+    volumes:
+      - $PWD:$PWD
+      - ./target/x86_64-unknown-linux-gnu/cargo/registry:/usr/local/cargo/registry
+      - ./target/x86_64-unknown-linux-gnu/cargo/git:/usr/local/cargo/git
+      - ./target/x86_64-unknown-linux-gnu/rustup/tmp:/usr/local/rustup/tmp
+    working_dir: $PWD
```
Comment (MOZGIII) [inline docker-compose.yml:39]:
```suggestion
    working_dir: $PWD
    user: $USER
```

Our builds can run as non-root now

File: docker-compose.yml [hunk 3]
```diff
@@ -213,14 +231,31 @@ services:
...
+      NATIVE_BUILD: "false"
+      TARGET: x86_64-unknown-linux-gnu
+      FEATURES: ""
+      CARGO_TERM_COLOR: always
+    volumes:
+      - $PWD:$PWD
+      - ./target/x86_64-unknown-linux-gnu/cargo/registry:/usr/local/cargo/registry
+      - ./target/x86_64-unknown-linux-gnu/cargo/git:/usr/local/cargo/git
+      - ./target/x86_64-unknown-linux-gnu/rustup/tmp:/usr/local/rustup/tmp
+    working_dir: $PWD
```
Comment (MOZGIII) [inline docker-compose.yml:248]:
```suggestion
    working_dir: $PWD
    user: $USER
```

**Comment by lukesteensen** [comment]
@MOZGIII the kubernetes integration test failures here look build-related. Is there something there we need to fix before merging this?

**Comment by MOZGIII** [comment]
Yes, the merge with master has to be corrected, I'll commit the patches here

**Comment by lukesteensen** [comment]
Great, thanks @MOZGIII! Feel free to go ahead and merge this once that's taken care of.

**Comment by MOZGIII** [comment]
Turn oot merge is ok, but the PR is unfinished. Nightly builds would fail, we need to somehow test them - or, like this time, we (would) miss the errors and break the nightly builds (if not k8s tests, which run through the docker build process).

**Pull request merged**